Implement ElementTypeToShortConverter. Second time.

aosyatnik opened the pull request:
This PR is copy of https://github.com/DynamoDS/Dynamo/pull/4417.

Reviewers
@Benglin , please take a look.

Link to YouTrack:
[MAGN-6197](http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-6197) DUI: Implement ElementTypeToShorthandConverter

aosyatnik commented on src/DynamoCoreWpf/Views/Search/LibraryView.xaml:
```diff
@@ -195,25 +195,66 @@
+                               Margin="0,0,10,0">
+                        <TextBlock.Style>
+                            <Style TargetType="{x:Type TextBlock}">
+                                <Setter Property="Visibility"
```
And in this PR I will answer about this setter.
If we just put Visibility in TextBlock e.g. `Visibility="Hidden"`. That will overwrite every `Setter`, that is part of Style. That's why if we want to use triggers, we should use setter in style.

pboyer commented:
Let's not append the "DUI" prefix to the PR names.  In the States, this is an acronym for "Driving Under the Influence" (of alcohol), which is not the association you want to make.  :)

aosyatnik commented:
I thought it's Dynamo UI ^_^

Benglin commented on src/DynamoCoreWpf/UI/Converters.cs:
```diff
@@ -2175,4 +2176,44 @@ public object Convert(object value, Type targetType, object parameter, CultureIn
+        {
+            var stringValue = value.ToString();
+            if (String.IsNullOrEmpty(stringValue))
+                throw new ArgumentException("Unknown element type.");
```
The `value` is already of type `ElementTypes`, there's no need to turn it into `string` and then back to `ElementTypes` again :)

Benglin commented:
Hi @aosyatnik, this works great (apart from the above point I mentioned). I would just like to see the region being enlarged to the whole *row* instead of just the text (if the mouse event does not get sent to you for other regions, perhaps you can use a non-transparent color so that it enables WPF hit-testing):

![limited-region-for-text](https://cloud.githubusercontent.com/assets/5086849/7678604/0e6675a4-fd88-11e4-9755-22202a9d05ab.png)

Thanks!

aosyatnik commented:
Hi, @Benglin .I've added opportunity to see label, when mouse is over expander button. But since I moved `TextBox` (ElementType TextBlock) level above in treeview, I have to add in `ElementTypeToShortConverter` more checks for flags.
Before ElementType TextBlock was at the hierarchical level. Hierarchical template level (e.g. root category) can have just one flag. But tree item level can be not only category, but also usual member (e.g. members of Migration category). These members can have several flags.

Benglin commented:
Thanks @aosyatnik for fixing them, LGTM!
